src/hooks/core/push-notification/usePushNotificationHook.ts

```diff
@@ -0,0 +1,218 @@
+import PushNotification from 'react-native-push-notification';
+import PushNotificationIOS from '@react-native-community/push-notification-ios';
+import messaging from '@react-native-firebase/messaging';
+import React, {useState} from 'react';
+import {Platform} from 'react-native';
+import {useNavigation} from '@react-navigation/core';
+import {useRecoilState} from 'recoil';
+
+import ChannelList from '../../../database/schema/ChannelListSchema';
+import profileAtom from '../../../atom/profileAtom';
+import useChatUtilsHook from '../chat/useChatUtilsHook';
+import useLocalDatabaseHook from '../../../database/hooks/useLocalDatabaseHook';
+import useUserAuthHook from '../auth/useUserAuthHook';
+import TokenStorage, {ITokenEnum} from '../../../utils/storage/custom/tokenStorage';
+import {getAnonymousUserId, getUserId} from '../../../utils/users';
+
+const usePushNotificationHook = () => {
+  const isIos = Platform.OS === 'ios';
+
+  const navigation = useNavigation();
+  const {signedProfileId, anonProfileId} = useUserAuthHook();
+  const {localDb} = useLocalDatabaseHook();
+  const {fetchChannelDetail, setSelectedChannel} = useChatUtilsHook();
+  const [, setProfileAtom] = useRecoilState(profileAtom);
+
+  const [isLoadingFetchingChannelDetail, setIsLoadingFetchingChannelDetail] = useState(false);
+
+  const __createChannel = () => {
+    PushNotification.createChannel(
+      {
+        channelId: 'bettersosialid', // (required)
+        channelName: 'New Messages & Comments', // (required)
+        playSound: true, // (optional) default: true
+        soundName: 'default', // (optional) See `soundName` parameter of `localNotification` function
+        importance: 4, // (optional) default: 4. Int value of the Android notification importance
+        vibrate: true // (optional) default: true. Creates the default vibration patten if true.
+      },
+      (created) => {
+        if (__DEV__) {
+          console.log(`createChannel returned '${created}'`);
+        }
+      } // (optional) callback returns whether the channel was created, false means it already existed.
+    );
+  };
+
+  const __pushNotifIos = (message) => {
+    const {title, body} = message.notification;
+    PushNotificationIOS.addNotificationRequest({
+      title,
+      body,
+      id: message.messageId,
+      userInfo: message.data
+    });
+  };
+
+  const __pushNotifAndroid = (remoteMessage) => {
+    const {title, body} = remoteMessage.notification;
+    PushNotification.localNotification({
+      id: '123',
+      title,
+      channelId: 'bettersosialid',
+      message: body,
+      data: remoteMessage.data
+    });
+  };
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

In the `__pushNotifIos` and `__pushNotifAndroid` functions, you are directly accessing properties from the `message` and `remoteMessage` objects without checking if they exist or not. This could potentially cause a crash if these properties do not exist. It would be better to use optional chaining (`?.`) to access these properties safely.

```diff
-    const {title, body} = message.notification;
+    const {title, body} = message?.notification;

-    const {title, body} = remoteMessage.notification;
+    const {title, body} = remoteMessage?.notification;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->